src/ui/map.js

```diff
@@ -1266,16 +1293,33 @@ class Map extends Camera {
      * });
      * @see [Example: Create a draggable point](https://docs.mapbox.com/mapbox-gl-js/example/drag-a-point/)
      */
-    off(type: MapEvent, layerId: any, listener: any) {
+    off(type: MapEvent, layerIds: any, listener: any) {
         if (listener === undefined) {
-            return super.off(type, layerId);
+            return super.off(type, layerIds);
+        }
+
+        if (!Array.isArray(layerIds)) {
+            layerIds = new Set([layerIds]);
+        } else {
+            layerIds = new Set(layerIds);
         }
 
-        const removeDelegatedListener = (delegatedListeners) => {
-            const listeners = delegatedListeners[type];
+        const areLayerArraysEqual = (hash1, hash2) => {
+            if (hash1.size !== hash2.size) {
+                return false; // at-least 1 arr has duplicate value(s)
+            }
+
+            // comparing values
+            for (const value of hash1) {
+                if (!hash2.has(value)) return false;
+            }
+            return true;
+        };
+
+        const removeDelegatedListener = (listeners: Array<any>) => {
```

Comment:
nit: since this is now always plural:

```suggestion
        const removeDelegatedListeners = (listeners: Array<any>) => {
```